### ISSUE TITLE
Metadata to indicate if fetch was instigated as a result of 103 early hints

### ISSUE THREAD

**Issue by LPardue** [w3c/webappsec-fetch-metadata#86]
[103 Early hints ](https://httpwg.org/specs/rfc8297.html) is seeing increased deployment and usage on the Internet. 

A server that sends too many hints is sub-optimal. A client has discretion about what hints it chooses to use and subsequently issue request for. From an observability perspective, it is difficult for operators to gain insight into whether a fetch was instigated directly because of an early hint, or for some other reason. That information could be used to tailor the set of hints.

The proposal therefore is to add a new indicator about whether a request is made due to early hints or not. This could be e.g. `Sec-Fetch-Early-Hint: ?1`, or some other format.

**Comment by arturjanc** [comment]
I wonder if it would make sense to use `Sec-Fetch-Mode` for that, e.g. `Sec-Fetch-Mode: early-hint`. If not, a separate header seems reasonable.

**Comment by LPardue** [comment]
Yeah thats a possibility, although the fact an early hint might have caused a fetch seems complimentary to the fetch mode? Not an expert on this though

**Comment by bashi** [comment]
Fetch mode of requests initiated by an early hints response can be same-origin, cors or no-cors, or even navigate if a client supports prefetch, so using `Sec-Fetch-Mode` may not be appropriate.

A request has an associated [initiator type](https://fetch.spec.whatwg.org/#request-initiator-type), the value of which is supposed to be 'early-hints' when the request was initiated by a 103 response. An option might be to expose it as `Sec-Fetch-Initiator-Type`, however, the definition of initiator type (and initiator) is somewhat ambigueous so I'm not sure that's a good idea. @yoavweiss

There is also [discussion](https://github.com/w3c/webappsec-fetch-metadata#do-we-need-both-initiator-and-destination) about whether we should include both initiator and destination.

**Comment by LPardue** [comment]
Oh, if initiator type of early hints was available via the Resource Timing API then that would probably work too https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-initiatortype

**Comment by arturjanc** [comment]
My guess is that if integrating with Resource Timing addresses this use case it might be a better solution because it avoids bytes to every network request. Otherwise, exposing the initiator type as `Sec-Fetch-Initiator-Type` seems okay, but given the overlap with `destination`, which all browsers are already sending (`Sec-Fetch-Dest`) it would result in a fair amount of duplication (and be somewhat confusing for developers).

**Comment by tunetheweb** [comment]
I think [it’s already exposed to ResourceTiming](https://github.com/w3c/resource-timing/issues/273) though IIRC there might be a bug with Chrome’s implementation of that.

I think @LPardue wants visibility from the server side though agree that does potentially create duplication.

**Comment by LPardue** [comment]
I could probably live Resource Timing but from https://github.com/w3c/resource-timing/issues/273 it seems there is no WPT to check whether it works or not? What would be the best way to look into that?

**Comment by yoavweiss** [comment]
Should we close this? It seems like RT addresses the use case
